File: 12_03_2022/Classify_Triplets.cpp
#include <iostream>
#include <math.h>

using namespace std;

int main(){

    int a,b;
    cin>>a>>b;

    for(int i=a;i<=b;i++){
        for(int j=i;j<=b;j++){
            for(int k=j;k<=b;k++){
                int x = pow(k,2)-pow(i,2)-pow(j,2);
                if(abs(x) == 0){
                    cout<<i<<" "<<j<<" "<<k<<" Exactly Pythagorean"<<endl;
                }
                else if(abs(x) == 1){
                    cout<<i<<" "<<j<<" "<<k<<" Very Close to Pythagorean"<<endl;
                }
                else if(abs(x) == 2){
                    cout<<i<<" "<<j<<" "<<k<<" Close to Pythagorean"<<endl;
                }
            }
        }
    }

    return 0;
}

File: March_2022/12_03_2022/Classify_Triplets.py
a = int(input())
b = int(input())

for i in range(a,b+1):
    for j in range(i,b+1):
        for k in range(j,b+1):
            x = k**2-i**2-j**2

            if(x == 0):
                print(i,j,k,"Exactly Pythagorean")
            elif(abs(x) == 1):
                print(i,j,k,"Very Close to Pythagorean")
            elif(abs(x) == 2):
                print(i,j,k,"Close to Pythagorean")